### PR TITLE
Feature/xml

### PR DESCRIPTION
**Pull request by staleyLANL**
Added some XML material. Work in progress.
@jlconlin Is this basically what you're looking for regarding style and layout?

### CODE REVIEW

**Comment by staleyLANL** [comment]
Oops, didn't do git commit... :-) Will do in a moment.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 __pycache__
 subprojects
 bin*
+*~
+build
```
Comment (jlconlin) [inline .gitignore]: Need to add `*.pyc` as well. We don't want our repository keeping those as they are automatically generated.

File: src/GNDStk/xml/test/doc/n-009_F_019.xml [hunk 2]
```diff
@@ -0,0 +1,8157 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (staleyLANL) [reply src/GNDStk/xml/test/doc/n-009_F_019.xml]: OK, I'll git ignore the *.pyc files.
Yes on feedback. I suppose the test code is worthwhile.
pugixml seems very easy to use, and is running fast here.
I'll change doc/ to resources/.

[the remark this answers is not in the document]
Comment (jlconlin) [reply src/GNDStk/xml/test/doc/n-009_F_019.xml]: You have to formally delete the *.pyc files (`git rm *.pyc` in whatever directories you’ve added them) in addition to ignoring them in the `.gitignore`.

**Comment by staleyLANL** [comment]
@jlconlin @whaeck Feel free to take a look at what I've been doing regarding generic document-tree access. This is still very much a work-in-progress. Files are too long and need to be substantially split up, and ideas need to be fleshed out more.

You've both spoken to me about what some people call the "builder" pattern, and about the Ranges library. Some of what I've done here is along some of the same lines, so I think you'll like it. *I* like it, at least. :-)

The latest work is something that I think will give us a working GNDS capability more quickly than my other approach of trying to work out all of the possible data structures. (But that other work can eventually complement this.)

For the new work, take a look specifically at the meta(), child(), and operator()() overloads in both class generic and nested class generic::node, in the header gnds.hpp. (I know, that file is way too long and should be split up.)  I also have knoop-based document and node classes to which similar stuff could be added, but I'm working on proof-of-principle now.

Then look at the new file, keyword.hpp, and you'll be able to see how the material in there interacts with the various function overloads I just spoke of. Note that I summarize some of the thinking in a comment at the top of that file.

Finally, take a look at gnds.cpp to see some concrete uses of the scheme I have in mind. (There's other, commented-out code in there, so the file is sort of cluttered. It needs to be split up as well.)

At present, the material in keyword.hpp, and the access examples in gnds.cpp, were done based on a GNDS file called a-002_He_004.xml. Obviously, there will be much more material to put in, in order to support the full range of possible GNDS constructs.

**Pull request closed without merging**